common/network-common/src/main/java/org/apache/spark/network/util/ByteArrayReadableChannel.java

```diff
@@ -19,44 +19,54 @@
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
 import java.nio.channels.ReadableByteChannel;
 
 import io.netty.buffer.ByteBuf;
 
 public class ByteArrayReadableChannel implements ReadableByteChannel {
   private ByteBuf data;
+  private boolean closed;
 
   public int readableBytes() {
-    return data.readableBytes();
+    return data == null ? 0 : data.readableBytes();
   }
 
-  public void feedData(ByteBuf buf) {
+  public void feedData(ByteBuf buf) throws ClosedChannelException {
+    if (closed) {
+      throw new ClosedChannelException();
+    }
+    if (data != null) {
+      throw new IllegalStateException("Already holding a ByteBuf");
+    }
     data = buf;
   }
 
   @Override
   public int read(ByteBuffer dst) throws IOException {
+    if (closed) {
```

Comment:
This is ok by itself, but I don't see any code calling `close()` so it's basically dead code...

Reply:
I can remove it if you like but that is basically the contract of a `ReadableByteChannel`.

Reply:
I think is fine. I took another look and `close()` is probably being called as part of closing the `CryptoInputStream`.